src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,115 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exeption.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private final Connection connection;
+
+    public BookDaoImpl(Connection connection) {
+        this.connection = connection;
+    }
+
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books "
+                + "(title, price) VALUES (?, ?)";
+        try (PreparedStatement preparedStatement
+                     = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            int executeUpdate = preparedStatement.executeUpdate();
+            if (executeUpdate < 1) {
+                throw new DataProcessingException("Can not process");
```

Comment:
The error message 'Can not process' is not very descriptive. Consider providing more context, such as 'Failed to insert book into database'.